src/utils/auth.js

```diff
@@ -0,0 +1,105 @@
+import { BASE_URL } from "../utils/constants";
+
+// Function to handle the server response
+export const handleServerResponse = (res) => {
+  return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+};
+
+// Sign up user (register)
+const signup = (name, email, password) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ name, email, password }), // Sending name, email, and password
+  }).then(handleServerResponse);
```

Comment:
**COULD BE IMPROVED**

You can make a special function for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(handleServerResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.